test/python/primitives/test_estimator.py

```diff
@@ -355,6 +355,27 @@ def get_op(i):
         keys = [_observable_key(get_op(i)) for i in range(5)]
         self.assertEqual(len(keys), len(set(keys)))
 
+    def test_reset(self):
+        """Test for circuits with reset."""
+        qc = QuantumCircuit(2)
+        qc.h(0)
+        qc.cx(0, 1)
+        qc.reset(0)
+        op = SparsePauliOp("ZI")
+
+        seed = 12
+        n = 250
+        with self.assertWarns(DeprecationWarning):
+            estimator = Estimator(options={"seed": seed})
+            result = estimator.run([qc for _ in range(n)], [op] * n).result()
+        # expectation values should be stochastic due to reset for subsystems
+        np.testing.assert_allclose(result.values.mean(), 0, atol=1e-1)
```

Comment:
If this runs very fast, could we increase `n` (maybe 1000)? If I did the math right this has still a ~1% chance of failing, which could be quite a lot in the long run. If it is not very fast I think we can also skip this test, as the behavior of `reset` should be tested in the `Statevector` class.

Reply:
n = 1000 takes around 0.3 sec on mac M1 (while n = 250 takes around 0.09 sec).
I will increase the number to n = 1000.

Reply:
We discussed with the team and the failure likelihood (now ~0.1%) is still too high as it would likely fail the CI once a day. Could you move that test to `test/randomized`? Then CI will not be blocked on the failures but we can still get a notice 🙂

Reply:
Because I fixed `seed`, it does not fail as far as there is no breaking change in `np.random` or `Statevector.reset`.

I printed `result.values.mean()` several times and the values are consistent.

```
(dev-qiskit) ima@mbp3 ~/t/4/q/terra (primitives/estimator-reset)> python -m unittest test/python/primitives/test_estimator.py test/python/primitives/test_statevector_estimator.py
.....0.006
0.032
........................0.006
0.03223976655569642
......
----------------------------------------------------------------------
Ran 35 tests in 1.942s

OK
(dev-qiskit) ima@mbp3 ~/t/4/q/terra (primitives/estimator-reset)> python -m unittest test/python/primitives/test_estimator.py test/python/primitives/test_statevector_estimator.py
.....0.006
0.032
........................0.006
0.03223976655569642
......
----------------------------------------------------------------------
Ran 35 tests in 1.838s

OK
(dev-qiskit) ima@mbp3 ~/t/4/q/terra (primitives/estimator-reset)> python -m unittest test/python/primitives/test_estimator.py test/python/primitives/test_statevector_estimator.py
.....0.006
0.032
........................0.006
0.03223976655569642
......
----------------------------------------------------------------------
Ran 35 tests in 1.834s

OK
```